torch/compiler/__init__.py

```diff
@@ -92,3 +101,64 @@ def disable(fn=None, recursive=True):
     import torch._dynamo
 
     return torch._dynamo.disable(fn, recursive)
+
+
+def dynamic_dim(t: torch.Tensor, index: int):
+    """
+    Note - [On Export Dynamic Dimension UX]
+
+    After a lot of discussion, we have settled on a dynamic marking API
```

Comment:
This docstring should be rewritten (at least the early part is very old). We have docs you can pull from. Also, document equality constraints.